internal/sql/pg/query.go

```diff
@@ -337,6 +337,26 @@ func decodeFromPGType(vals ...any) ([]any, error) {
 		switch v := v.(type) {
 		default:
 			return v, nil
+
+		// we need to handle all ints as int64 since Kwil treats all
+		// ints as int64, but an integer literal in postgres can get
+		// returned as an int32
+		case int:
+			return int64(v), nil
+		case int8:
+			return int64(v), nil
+		case int16:
+			return int64(v), nil
+		case int32:
+			return int64(v), nil
+		case int64:
+			return v, nil
+		case uint:
+			return int64(v), nil
+		case uint16:
```

Comment:
Did you want `uint8` here as well?  (although I don't think postgres+pgx is ever going to use an unsigned type, but I could be wrong).

Also, I'm wondering if this forced int64 behavior is to opinionated for this level of the system.  e.g. is this an engine requirement?  Should it be forced on all consumers of pg including internal stores, etc?  In engine, we have an `cleanseIntValue` function that does very similar.  Perhaps we should be applying this type issue up there somewhere.

Reply:
I intentionally did not include uint8, in case it is reading bytes

Reply:
> Also, I'm wondering if this forced int64 behavior is to opinionated for this level of the system

Very much could be, however the SQL package is already familiar with the other Kwil types, and does not support other types like float, which is already pretty Kwil specific. But if you think it is too opinionated, I am definitely ok to remove

Reply:
OK.  Regardless of "the right place" for this.  It's hard to see why it would be a problem for any lesser integer types to return to `query` consumers as `int64`, so OK. :heavy_check_mark:

Reply:
> and does not support other types like float, which is already pretty Kwil specific

Well, it returns float unconverted if you're using `query`.  But NBD.  Nothing is losing precision.